bindings/matrix-sdk-ffi/src/sliding_sync.rs

```diff
@@ -652,35 +664,43 @@ impl SlidingSync {
         let inner = self.inner.clone();
         let client = self.client.clone();
         let observer = self.observer.clone();
+        let stop_loop = Arc::new(AtomicBool::new(false));
+        let remote_stopper = stop_loop.clone();
 
-        Arc::new(
-            RUNTIME
-                .spawn(async move {
-                    let stream = inner.stream().await.unwrap();
-                    pin_mut!(stream);
-                    loop {
-                        let update = match stream.next().await {
-                            Some(Ok(u)) => u,
-                            Some(Err(e)) => {
-                                if client.process_sync_error(e) == LoopCtrl::Break {
-                                    warn!("loop was stopped by client error processing");
-                                    break;
-                                } else {
-                                    continue;
-                                }
-                            }
-                            None => {
-                                warn!("Inner streaming loop ended unexpectedly");
-                                break;
-                            }
-                        };
-                        if let Some(ref observer) = *observer.read().unwrap() {
-                            observer.did_receive_sync_update(update.into());
+        let stoppable = Arc::new(StoppableSpawn::with_callback(Box::new(move || {
+            remote_stopper.store(true, Ordering::Relaxed);
```

Comment:
We're still using `Relaxed` here and in a couple of other places, didn't we want to use `SeqCst`?